pkg/autodiscovery/providers/env.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package providers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+const name = "dd_logs_config_custom_config"
+
+var collectors = []func(*EnvVarProvider) ([]integration.Config, error){
```

Comment:
why not `[]func() ([]integration.Config, error)` ?

Reply:
nits `var collectors = []func(*EnvVarProvider) ([]integration.Config, error) {`

Reply:
otherwise collector functions should be static, they don't need to be per provider I think, right ?

Reply:
Yes, I agree, they don't use the EnvVarProvider